x/gamm/pool-models/balancer/pool.go

```diff
@@ -663,6 +663,19 @@ func (p *Pool) JoinPool(ctx sdk.Context, tokensIn sdk.Coins, swapFee sdk.Dec) (n
 	return numShares, nil
 }
 
+// JoinPoolNoSwap calculates the number of shares needed for an all-asset join given tokensIn with swapFee applied.
+// It updates the liquidity if the pool is joined successfully. If not, returns error.
+func (p *Pool) JoinPoolNoSwap(ctx sdk.Context, tokensIn sdk.Coins, swapFee sdk.Dec) (numShares sdk.Int, err error) {
+	numShares, newLiquidity, _, err := p.CalcJoinPoolNoSwapShares(ctx, tokensIn, swapFee)
```

Comment:
Should we rename this similar to the name of the other caller?

```suggestion
	numShares, tokensJoined, _, err := p.CalcJoinPoolNoSwapShares(ctx, tokensIn, swapFee)
```